SpeechResponder/CopyPersonalityWindow.xaml.cs

```diff
@@ -1,52 +1,122 @@
+
     /// <summary>
     /// Interaction logic for CopyPersonalityWindow.xaml
     /// </summary>
-    public partial class CopyPersonalityWindow : Window, INotifyPropertyChanged
+    public partial class CopyPersonalityWindow : Window, INotifyPropertyChanged, IDataErrorInfo
     {
-        private Personality personality;
+        private readonly HashSet<string> existingNames = new HashSet<string>();
 
         private string personalityName;
         public string PersonalityName
         {
-            get { return personalityName; }
-            set { personalityName = value; OnPropertyChanged("PersonalityName"); }
+            get => personalityName;
+            set
+            {
+                personalityName = value;
+                OnPropertyChanged(PersonalityName);
+            }
         }
+
         private string personalityDescription;
         public string PersonalityDescription
         {
-            get { return personalityDescription; }
-            set { personalityDescription = value; OnPropertyChanged("PersonalityDescription"); }
+            get => personalityDescription;
+            set
+            {
+                personalityDescription = value; 
+                OnPropertyChanged(PersonalityDescription);
+            }
+        }
+
+        bool CanAccept()
+        {
+            return string.IsNullOrWhiteSpace(Error);
         }
 
         public event PropertyChangedEventHandler PropertyChanged;
-        protected void OnPropertyChanged(string name)
+        private void OnPropertyChanged(string name)
         {
             PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(name));
+            acceptButton.IsEnabled = CanAccept();
         }
 
-        public CopyPersonalityWindow(Personality personality)
+        public CopyPersonalityWindow(IEnumerable<Personality> personalities)
         {
             InitializeComponent();
             DataContext = this;
+            foreach (var personality in personalities)
+            {
+                existingNames.Add(personality.Name.ToLower());
+            }
 
-            this.personality = personality;
+            acceptButton.IsEnabled = CanAccept();
         }
 
         private void acceptButtonClick(object sender, RoutedEventArgs e)
         {
-            DialogResult = (PersonalityName != null && PersonalityName.Trim() != "");
-            this.Close();
+            DialogResult = !Validation.GetHasError(PersonalityNameField);
+            Close();
         }
 
         private void cancelButtonClick(object sender, RoutedEventArgs e)
         {
             DialogResult = false;
-            this.Close();
+            Close();
+        }
+
+#region DataErrorInfo
+        // C#'s lack of string-based enums is annoying here. We have to resort to string literals, which are vulnerable to typos.
+        public string Error
+        {
+            get
+            {
+                string result = this["PersonalityName"];
```

Comment:
Might use `nameof(PersonalityName)` rather than "PersonalityName" to increase resiliency in case variables are later renamed.

Reply:
Good idea, thanks.